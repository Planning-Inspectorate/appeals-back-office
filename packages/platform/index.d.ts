import { HandlerFunction } from 'got';

declare module '@pins/platform' {
	export function createTtlHandler(): HandlerFunction;
	export function validatePostcode(value: string): boolean;
	export function validatePastDate(value: Date | string | number): boolean;
	export function validateFutureDate(value: Date | string | number): boolean;
	export function yesterday(): Date;
}

export interface LocalPlanningDepartment {
	FID: number;
	LPA21CD: string;
	LPA21NM: string;
	Co_terminous: 'Y' | 'N';
}
