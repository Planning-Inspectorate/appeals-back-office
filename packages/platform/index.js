export * from './http/ttl.js';
export * from './util/date.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
