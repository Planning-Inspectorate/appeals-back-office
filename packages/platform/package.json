{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=22.0.0 <= 22.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "test": "cross-env NODE_ENV=test NODE_OPTIONS=\"--experimental-vm-modules --expose-gc\" npx jest --logHeapUsage --ci",
    "test:cov": "cross-env NODE_ENV=test NODE_OPTIONS=\"--experimental-vm-modules --expose-gc\" npx jest --logHeapUsage --ci --coverage",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.3",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.4",
    "@types/lodash-es": "^4.17.12",
    "cross-env": "^7.0.3",
    "jest": "^29.7.0",
    "type-fest": "^4.37.0"
  }
}
