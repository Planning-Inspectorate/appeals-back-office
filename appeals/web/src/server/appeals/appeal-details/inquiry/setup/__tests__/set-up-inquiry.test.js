// @ts-nocheck
import { appealData } from '#testing/app/fixtures/referencedata.js';
import { createTestEnvironment } from '#testing/index.js';
import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { behavesLikeAddressForm } from '#testing/app/shared-examples/address-form.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);
const baseUrl = '/appeals-service/appeal-details';

describe('set up inquiry', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	describe('GET /inquiry/setup', () => {
		// eslint-disable-next-line jest/expect-expect
		it('should redirect to /inquiry/setup/date', () => {
			return new Promise((resolve) => {
				request
					.get(`${baseUrl}/2/inquiry/setup`)
					.expect(302)
					.expect('Location', `${baseUrl}/2/inquiry/setup/date`)
					.end(resolve);
			});
		});
	});

	describe('GET /inquiry/setup/date', () => {
		const appealId = 2;

		let pageHtml;

		beforeAll(async () => {
			nock('http://test/')
				.get(`/appeals/${appealId}`)
				.reply(200, { ...appealData, appealId });

			const response = await request.get(`${baseUrl}/${appealId}/inquiry/setup/date`);
			pageHtml = parseHtml(response.text);
		});

		it('should match the snapshot', () => {
			expect(pageHtml.innerHTML).toMatchSnapshot();
		});

		it('should render the correct heading', () => {
			expect(pageHtml.querySelector('h1')?.innerHTML.trim()).toBe('Inquiry date and time');
		});

		it('should render a Date field', () => {
			expect(pageHtml.querySelector('input#inquiry-date-day')).not.toBeNull();
			expect(pageHtml.querySelector('input#inquiry-date-month')).not.toBeNull();
			expect(pageHtml.querySelector('input#inquiry-date-year')).not.toBeNull();
		});

		it('should render a Time field', () => {
			expect(pageHtml.querySelector('input#inquiry-time-hour')).not.toBeNull();
			expect(pageHtml.querySelector('input#inquiry-time-minute')).not.toBeNull();
		});

		it('should render any saved response', async () => {
			nock('http://test/')
				.get(`/appeals/${appealId}`)
				.twice()
				.reply(200, { ...appealData, appealId });

			//set session data with post request
			await request.post(`${baseUrl}/${appealId}/inquiry/setup/date`).send({
				'inquiry-date-day': '01',
				'inquiry-date-month': '02',
				'inquiry-date-year': '3025',
				'inquiry-time-hour': '12',
				'inquiry-time-minute': '00'
			});

			const response = await request.get(`${baseUrl}/${appealId}/inquiry/setup/date`);

			pageHtml = parseHtml(response.text);

			expect(pageHtml.querySelector('input#inquiry-date-day').getAttribute('value')).toEqual('01');
			expect(pageHtml.querySelector('input#inquiry-date-month').getAttribute('value')).toEqual(
				'02'
			);
			expect(pageHtml.querySelector('input#inquiry-date-year').getAttribute('value')).toEqual(
				'3025'
			);
			expect(pageHtml.querySelector('input#inquiry-time-hour').getAttribute('value')).toEqual('12');
			expect(pageHtml.querySelector('input#inquiry-time-minute').getAttribute('value')).toEqual(
				'00'
			);
		});
	});

	describe('POST /inquiry/setup/date', () => {
		const appealId = 2;

		beforeEach(() => {
			nock('http://test/')
				.get(`/appeals/${appealId}`)
				.reply(200, { ...appealData, appealId });
		});

		it('should redirect to /inquiry/setup/address with valid inputs', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/inquiry/setup/date`).send({
				'inquiry-date-day': '01',
				'inquiry-date-month': '02',
				'inquiry-date-year': '3025',
				'inquiry-time-hour': '12',
				'inquiry-time-minute': '00'
			});

			expect(response.statusCode).toBe(302);
			expect(response.headers.location).toBe(`${baseUrl}/${appealId}/inquiry/setup/estimation`);
		});

		it('should return 400 on invalid date with appropriate error message', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/inquiry/setup/date`).send({
				'inquiry-date-day': '31',
				'inquiry-date-month': '02',
				'inquiry-date-year': '2025',
				'inquiry-time-hour': '12',
				'inquiry-time-minute': '00'
			});

			expect(response.statusCode).toBe(400);

			const errorSummaryHtml = parseHtml(response.text, {
				rootElement: '.govuk-error-summary',
				skipPrettyPrint: true
			}).innerHTML;

			expect(errorSummaryHtml).toContain('There is a problem</h2>');
			expect(errorSummaryHtml).toContain('Inquiry date must be a real date');
		});

		it('should return 400 on missing date with appropriate error message', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/inquiry/setup/date`).send({
				'inquiry-time-hour': '12',
				'inquiry-time-minute': '00'
			});

			expect(response.statusCode).toBe(400);

			const errorSummaryHtml = parseHtml(response.text, {
				rootElement: '.govuk-error-summary',
				skipPrettyPrint: true
			}).innerHTML;

			expect(errorSummaryHtml).toContain('There is a problem</h2>');
			expect(errorSummaryHtml).toContain('Enter the inquiry date');
		});

		it('should return 400 on date in the past with appropriate error message', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/inquiry/setup/date`).send({
				'inquiry-date-day': '28',
				'inquiry-date-month': '02',
				'inquiry-date-year': '1999',
				'inquiry-time-hour': '12',
				'inquiry-time-minute': '00'
			});

			expect(response.statusCode).toBe(400);

			const errorSummaryHtml = parseHtml(response.text, {
				rootElement: '.govuk-error-summary',
				skipPrettyPrint: true
			}).innerHTML;

			expect(errorSummaryHtml).toContain('There is a problem</h2>');
			expect(errorSummaryHtml).toContain('The inquiry date must be in the future');
		});

		it('should return 400 on invalid time with appropriate error message', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/inquiry/setup/date`).send({
				'inquiry-date-day': '28',
				'inquiry-date-month': '02',
				'inquiry-date-year': '3025',
				'inquiry-time-hour': '99',
				'inquiry-time-minute': '99'
			});

			expect(response.statusCode).toBe(400);

			const errorSummaryHtml = parseHtml(response.text, {
				rootElement: '.govuk-error-summary',
				skipPrettyPrint: true
			}).innerHTML;

			expect(errorSummaryHtml).toContain('There is a problem</h2>');
			expect(errorSummaryHtml).toContain('Enter a real inquiry time');
		});

		it('should return 400 on missing time with appropriate error message', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/inquiry/setup/date`).send({
				'inquiry-date-day': '28',
				'inquiry-date-month': '02',
				'inquiry-date-year': '3025'
			});

			expect(response.statusCode).toBe(400);

			const errorSummaryHtml = parseHtml(response.text, {
				rootElement: '.govuk-error-summary',
				skipPrettyPrint: true
			}).innerHTML;

			expect(errorSummaryHtml).toContain('There is a problem</h2>');
			expect(errorSummaryHtml).toContain('Enter the inquiry time');
		});
	});

	describe('GET /inquiry/setup/estimation', () => {
		const appealId = 2;

		let pageHtml;

		beforeAll(async () => {
			nock('http://test/')
				.get(`/appeals/${appealId}`)
				.twice()
				.reply(200, { ...appealData, appealId });

			// set session data with post request
			await request.post(`${baseUrl}/${appealId}/Inquiry/setup/estimation`).send({
				inquiryEstimationYesNo: 'yes'
			});

			const response = await request.get(`${baseUrl}/${appealId}/Inquiry/setup/estimation`);
			pageHtml = parseHtml(response.text);
		});

		it('should match the snapshot', () => {
			expect(pageHtml.innerHTML).toMatchSnapshot();
		});

		it('should render the correct heading', () => {
			expect(pageHtml.querySelector('h1')?.innerHTML.trim()).toBe(
				'Do you know the expected number of days to carry out the inquiry?'
			);
		});

		it('should render a radio button for inquiry estimation', () => {
			expect(pageHtml.querySelector('input[name="inquiryEstimationYesNo"]')).not.toBeNull();
		});

		it('should render an input for inquiry estimation days', () => {
			expect(pageHtml.querySelector('input[name="inquiryEstimationDays"]')).not.toBeNull();
		});
	});

	describe('POST /inquiry/setup/estimation', () => {
		const appealId = 2;

		beforeEach(() => {
			nock('http://test/')
				.get(`/appeals/${appealId}`)
				.reply(200, { ...appealData, appealId });
		});

		it('should redirect to /Inquiry/setup/address when answering no', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/Inquiry/setup/estimation`).send({
				inquiryEstimationYesNo: 'no'
			});

			expect(response.statusCode).toBe(302);
			expect(response.headers.location).toBe(`${baseUrl}/${appealId}/inquiry/setup/address`);
		});

		it('should redirect to /Inquiry/setup/address when answering yes', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/Inquiry/setup/estimation`).send({
				inquiryEstimationYesNo: 'yes',
				inquiryEstimationDays: 12
			});

			expect(response.statusCode).toBe(302);
			expect(response.headers.location).toBe(`${baseUrl}/${appealId}/inquiry/setup/address`);
		});

		it('should return 400 on missing inquiryEstimationYesNo with appropriate error message', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/Inquiry/setup/address`).send({});

			expect(response.statusCode).toBe(400);

			const errorSummaryHtml = parseHtml(response.text, {
				rootElement: '.govuk-error-summary',
				skipPrettyPrint: true
			}).innerHTML;

			expect(errorSummaryHtml).toContain('There is a problem</h2>');
			expect(errorSummaryHtml).toContain(
				'Select yes if you know the address of where the inquiry will take place'
			);
		});
	});

	describe('GET /inquiry/setup/address', () => {
		const appealId = 2;

		let pageHtml;

		beforeAll(async () => {
			nock('http://test/')
				.get(`/appeals/${appealId}`)
				.twice()
				.reply(200, { ...appealData, appealId });

			// set session data with post request
			await request.post(`${baseUrl}/${appealId}/Inquiry/setup/address`).send({
				addressKnown: 'yes'
			});

			const response = await request.get(`${baseUrl}/${appealId}/Inquiry/setup/address`);
			pageHtml = parseHtml(response.text);
		});

		it('should match the snapshot', () => {
			expect(pageHtml.innerHTML).toMatchSnapshot();
		});

		it('should render the correct heading', () => {
			expect(pageHtml.querySelector('h1')?.innerHTML.trim()).toBe(
				'Do you know the address of where the inquiry will take place?'
			);
		});

		it('should render a radio button for address known', () => {
			expect(pageHtml.querySelector('input[name="addressKnown"]')).not.toBeNull();
		});

		it('should render a radio button for address unknown', () => {
			expect(pageHtml.querySelector('input[name="addressKnown"]')).not.toBeNull();
		});

		it('should check the submitted value', () => {
			expect(
				pageHtml.querySelector('input[name="addressKnown"][value="yes"]')?.getAttribute('checked')
			).toBeDefined();
		});
	});

	describe('POST /inquiry/setup/address', () => {
		const appealId = 2;

		beforeEach(() => {
			nock('http://test/')
				.get(`/appeals/${appealId}`)
				.reply(200, { ...appealData, appealId });
		});

		it('should redirect to /inquiry/setup/timetable-due-dates when answering no', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/Inquiry/setup/address`).send({
				addressKnown: 'no'
			});

			expect(response.statusCode).toBe(302);
			expect(response.headers.location).toBe(
				`${baseUrl}/${appealId}/inquiry/setup/timetable-due-dates`
			);
		});

		it('should redirect to /Inquiry/setup/address-details when answering yes', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/Inquiry/setup/address`).send({
				addressKnown: 'yes'
			});

			expect(response.statusCode).toBe(302);
			expect(response.headers.location).toBe(
				`${baseUrl}/${appealId}/inquiry/setup/address-details`
			);
		});

		it('should return 400 on missing addressKnown with appropriate error message', async () => {
			const response = await request.post(`${baseUrl}/${appealId}/Inquiry/setup/address`).send({});

			expect(response.statusCode).toBe(400);

			const errorSummaryHtml = parseHtml(response.text, {
				rootElement: '.govuk-error-summary',
				skipPrettyPrint: true
			}).innerHTML;

			expect(errorSummaryHtml).toContain('There is a problem</h2>');
			expect(errorSummaryHtml).toContain(
				'Select yes if you know the address of where the inquiry will take place'
			);
		});
	});

	describe('GET /inquiry/setup/address-details', () => {
		const appealId = 2;

		let pageHtml;

		beforeAll(async () => {
			nock('http://test/')
				.get(`/appeals/${appealId}`)
				.reply(200, { ...appealData, appealId });

			const response = await request.get(`${baseUrl}/${appealId}/Inquiry/setup/address-details`);
			pageHtml = parseHtml(response.text);
		});

		it('should match the snapshot', () => {
			expect(pageHtml.innerHTML).toMatchSnapshot();
		});

		it('should render the correct heading', () => {
			expect(pageHtml.querySelector('h1')?.innerHTML.trim()).toBe('Inquiry address');
		});

		it('should render a text input for address line 1', () => {
			expect(pageHtml.querySelector('input[name="addressLine1"]')).not.toBeNull();
		});

		it('should render a text input for address line 2', () => {
			expect(pageHtml.querySelector('input[name="addressLine2"]')).not.toBeNull();
		});

		it('should render a text input for town', () => {
			expect(pageHtml.querySelector('input[name="town"]')).not.toBeNull();
		});

		it('should render a text input for county', () => {
			expect(pageHtml.querySelector('input[name="county"]')).not.toBeNull();
		});

		it('should render a text input for postcode', () => {
			expect(pageHtml.querySelector('input[name="postCode"]')).not.toBeNull();
		});
	});

	describe('POST /inquiry/setup/address-details', () => {
		const appealId = 2;

		beforeEach(() => {
			nock('http://test/')
				.get(`/appeals/${appealId}`)
				.reply(200, { ...appealData, appealId });
		});

		it('should redirect to /Inquiry/setup/timetable-due-dates with valid inputs', async () => {
			const response = await request
				.post(`${baseUrl}/${appealId}/Inquiry/setup/address-details`)
				.send({
					addressLine1: 'Flat 9',
					addressLine2: '123 Gerbil Drive',
					town: 'Blarberton',
					county: 'Slabshire',
					postCode: 'X25 3YZ'
				});

			expect(response.statusCode).toBe(302);
			expect(response.headers.location).toBe(
				`${baseUrl}/${appealId}/inquiry/setup/timetable-due-dates`
			);
		});

		behavesLikeAddressForm({
			request,
			url: `${baseUrl}/${appealId}/Inquiry/setup/address-details`
		});
	});
});
