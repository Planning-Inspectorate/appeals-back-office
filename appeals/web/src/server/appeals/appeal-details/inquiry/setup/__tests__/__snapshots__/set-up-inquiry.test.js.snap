// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`set up inquiry GET /Inquiry/setup/address should match the snapshot 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds"><span class="govuk-caption-l">Appeal 351062 - start case</span>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full">
                    <form method="POST" novalidate="novalidate">
                        <div class="govuk-form-group">
                            <fieldset class="govuk-fieldset">
                                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
                                    <h1 class="govuk-fieldset__heading"> Do you know the address of where the inquiry will take place?</h1>
                                </legend>
                                <div class="govuk-radios" data-module="govuk-radios">
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="address-known" name="addressKnown"
                                        type="radio" value="yes" checked>
                                        <label class="govuk-label govuk-radios__label" for="address-known">Yes</label>
                                    </div>
                                    <div class="govuk-radios__item">
                                        <input class="govuk-radios__input" id="address-known-2" name="addressKnown"
                                        type="radio" value="no">
                                        <label class="govuk-label govuk-radios__label" for="address-known-2">No</label>
                                    </div>
                                </div>
                            </fieldset>
                        </div>
                        <button type="submit" data-prevent-double-click="true" class="govuk-button"
                        data-module="govuk-button">Continue</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
</main>"
`;

exports[`set up inquiry GET /Inquiry/setup/address-details should match the snapshot 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds"><span class="govuk-caption-l">Appeal 351062</span>
            <h1 class="govuk-heading-l">Inquiry address</h1>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full">
                    <form method="POST" novalidate="novalidate">
                        <div class="govuk-form-group">
                            <label class="govuk-label" for="address-line-1">Address line 1</label>
                            <input class="govuk-input" id="address-line-1" name="addressLine1"
                            type="text" value="">
                        </div>
                        <div class="govuk-form-group">
                            <label class="govuk-label" for="address-line-2">Address line 2 (optional)</label>
                            <input class="govuk-input" id="address-line-2"
                            name="addressLine2" type="text" value="">
                        </div>
                        <div class="govuk-form-group">
                            <label class="govuk-label" for="town">Town or city</label>
                            <input class="govuk-input govuk-input govuk-input--width-20"
                            id="town" name="town" type="text" value="">
                        </div>
                        <div class="govuk-form-group">
                            <label class="govuk-label" for="county">County (optional)</label>
                            <input class="govuk-input govuk-input govuk-input--width-20"
                            id="county" name="county" type="text" value="">
                        </div>
                        <div class="govuk-form-group">
                            <label class="govuk-label" for="post-code">Postcode</label>
                            <input class="govuk-input govuk-input govuk-input--width-10"
                            id="post-code" name="postCode" type="text" value="">
                        </div>
                        <div class="govuk-form-group">
                            <input class="govuk-input" id="operation-type" name="operationType" type="hidden"
                            value="">
                        </div>
                        <button type="submit" data-prevent-double-click="true" class="govuk-button"
                        data-module="govuk-button">Continue</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
</main>"
`;

exports[`set up inquiry GET /inquiry/setup/date should match the snapshot 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds"><span class="govuk-caption-l">Appeal 351062 - start case</span>
            <h1 class="govuk-heading-l">Inquiry date and time</h1>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full">
                    <form method="POST" novalidate="novalidate">
                        <div class="govuk-form-group">
                            <fieldset class="govuk-fieldset" role="group" aria-describedby="inquiry-date-hint">
                                <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">Date</legend>
                                <div id="inquiry-date-hint" class="govuk-hint">For example, 31 3 2025</div>
                                <div class="govuk-date-input" id="inquiry-date">
                                    <div class="govuk-date-input__item">
                                        <div class="govuk-form-group">
                                            <label class="govuk-label govuk-date-input__label" for="inquiry-date-day">Day</label>
                                            <input class="govuk-input govuk-date-input__input govuk-input--width-2 govuk-input govuk-date-input__input"
                                            id="inquiry-date-day" name="inquiry-date-day" type="text" value="" inputmode="numeric">
                                        </div>
                                    </div>
                                    <div class="govuk-date-input__item">
                                        <div class="govuk-form-group">
                                            <label class="govuk-label govuk-date-input__label" for="inquiry-date-month">Month</label>
                                            <input class="govuk-input govuk-date-input__input govuk-input--width-2 govuk-input govuk-date-input__input"
                                            id="inquiry-date-month" name="inquiry-date-month" type="text" value=""
                                            inputmode="numeric">
                                        </div>
                                    </div>
                                    <div class="govuk-date-input__item">
                                        <div class="govuk-form-group">
                                            <label class="govuk-label govuk-date-input__label" for="inquiry-date-year">Year</label>
                                            <input class="govuk-input govuk-date-input__input govuk-input--width-4 govuk-input govuk-date-input__input"
                                            id="inquiry-date-year" name="inquiry-date-year" type="text" value="" inputmode="numeric">
                                        </div>
                                    </div>
                                </div>
                            </fieldset>
                        </div>
                        <fieldset class="govuk-fieldset">
                            <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">Time</legend>
                            <div id="hearing-date-hint" class="govuk-hint">For example, 9:00 or 13:15</div>
                            <div class="govuk-date-input">
                                <div class="govuk-date-input__item govuk-!-margin-right-1">
                                    <div class="govuk-form-group">
                                        <label class="govuk-label" for="inquiry-time-hour">Hour</label>
                                        <input class="govuk-input govuk-input--width-2" id="inquiry-time-hour"
                                        name="inquiry-time-hour" type="text" value="10">
                                    </div>
                                </div>
                                <div class="govuk-date-input__item govuk-!-font-size-19 govuk-!-padding-0 govuk-!-margin-right-1">:</div>
                                <div class="govuk-date-input__item">
                                    <div class="govuk-form-group">
                                        <label class="govuk-label" for="inquiry-time-minute">Minute</label>
                                        <input class="govuk-input govuk-input--width-2" id="inquiry-time-minute"
                                        name="inquiry-time-minute" type="text" value="00">
                                    </div>
                                </div>
                            </div>
                        </fieldset>
                        <button type="submit" data-prevent-double-click="true" class="govuk-button"
                        data-module="govuk-button">Continue</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
</main>"
`;

exports[`set up inquiry POST /Inquiry/setup/address-details should re-render the page if addressLine1 is an empty string 1`] = `
"<div class="govuk-error-summary" data-module="govuk-error-summary">
    <div role="alert">
        <h2 class="govuk-error-summary__title"> There is a problem</h2>
        <div class="govuk-error-summary__body">
            <ul class="govuk-list govuk-error-summary__list">
                <li><a href="#address-line-1">Enter address line 1</a>
                </li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`set up inquiry POST /Inquiry/setup/address-details should re-render the page if addressLine1 is null 1`] = `
"<div class="govuk-error-summary" data-module="govuk-error-summary">
    <div role="alert">
        <h2 class="govuk-error-summary__title"> There is a problem</h2>
        <div class="govuk-error-summary__body">
            <ul class="govuk-list govuk-error-summary__list">
                <li><a href="#address-line-1">Enter address line 1</a>
                </li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`set up inquiry POST /Inquiry/setup/address-details should re-render the page if the postcode is an empty string 1`] = `
"<div class="govuk-error-summary" data-module="govuk-error-summary">
    <div role="alert">
        <h2 class="govuk-error-summary__title"> There is a problem</h2>
        <div class="govuk-error-summary__body">
            <ul class="govuk-list govuk-error-summary__list">
                <li><a href="#post-code">Enter postcode</a>
                </li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`set up inquiry POST /Inquiry/setup/address-details should re-render the page if the postcode is invalid 1`] = `
"<div class="govuk-error-summary" data-module="govuk-error-summary">
    <div role="alert">
        <h2 class="govuk-error-summary__title"> There is a problem</h2>
        <div class="govuk-error-summary__body">
            <ul class="govuk-list govuk-error-summary__list">
                <li><a href="#post-code">Enter a full UK postcode</a>
                </li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`set up inquiry POST /Inquiry/setup/address-details should re-render the page if the postcode is null 1`] = `
"<div class="govuk-error-summary" data-module="govuk-error-summary">
    <div role="alert">
        <h2 class="govuk-error-summary__title"> There is a problem</h2>
        <div class="govuk-error-summary__body">
            <ul class="govuk-list govuk-error-summary__list">
                <li><a href="#post-code">Enter postcode</a>
                </li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`set up inquiry POST /Inquiry/setup/address-details should re-render the page if town is an empty string 1`] = `
"<div class="govuk-error-summary" data-module="govuk-error-summary">
    <div role="alert">
        <h2 class="govuk-error-summary__title"> There is a problem</h2>
        <div class="govuk-error-summary__body">
            <ul class="govuk-list govuk-error-summary__list">
                <li><a href="#town">Enter town or city</a>
                </li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`set up inquiry POST /Inquiry/setup/address-details should re-render the page if town is null 1`] = `
"<div class="govuk-error-summary" data-module="govuk-error-summary">
    <div role="alert">
        <h2 class="govuk-error-summary__title"> There is a problem</h2>
        <div class="govuk-error-summary__body">
            <ul class="govuk-list govuk-error-summary__list">
                <li><a href="#town">Enter town or city</a>
                </li>
            </ul>
        </div>
    </div>
</div>"
`;
